andela/ah-infinity-stones
# #161966920 Implement Article Like and Dislike

#### What does this PR do?
- Add article and dislike feature
#### Description of Task to be completed?
- Update like-dislike model
- Implement like-dislike endpoint
- Implement tests
#### How should this be manually tested?
1. `git clone -b ft-like-and-dislike-articles-161966920 https://github.com/andela/ah-infinity-stones` and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_NAME='ah_dev'
export DATABASE_USER='user'
export DATABASE_PASSWORD=pass'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'
export EMAIL_HOST_USER='username'
export EMAIL_HOST_PASSWORD='pass'
export SECRET_KEY='key-is-secret'
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/users/` through [Postman](https://www.getpostman.com/), provide inputs in the following format and create account.
```
{ "user": {
"username": "username44",
"email": "username@user.com",
"password": "@Password321"	
}
}
```
Check your email and activate account. Use the registration token to acess the following endpoints:
- `http://127.0.0.1:8000/api/articles/` - Post an article
- `http://127.0.0.1:8000/api/articles/art_slug/like-dislike` -  like or dislike the article created.

8. Run tests`python3 manage.py test authors.apps.articles.tests.test_like_article`
#### What are the relevant pivotal tracker stories?
[#161966920](https://www.pivotaltracker.com/story/show/161966920)
#### Screenshots
<img width="977" alt="screenshot 2018-12-18 at 20 40 58" src="https://user-images.githubusercontent.com/12385482/50172105-4fcc0000-0305-11e9-93bb-ae792569637c.png">
<img width="975" alt="screenshot 2018-12-18 at 20 41 16" src="https://user-images.githubusercontent.com/12385482/50172118-578ba480-0305-11e9-9a5c-969ed74b33d4.png">
